### ISSUE TITLE
Leaking FakeSlant definitions

### ISSUE THREAD

**Issue by wspr** [wspr/fontspec#193]
[Oops](http://tex.stackexchange.com/questions/228831/is-there-a-good-workaround-for-the-fact-that-when-using-fontspec-to-do-fake-slan). Minimal example:
```
\documentclass{memoir}
\usepackage{fontspec}
\begin{document}
\setmainfont{Latin Modern Roman}[
  SlantedFont={Linux Libertine O},
  SlantedFeatures={FakeSlant=0.5},
]
Small Caps is slanted now: \textsc{FOO} and also slanted is slanted: \textsl{FOO} 
\end{document}
```

**Comment by itoijala** [comment]
The problem seems to not be limited to `FakeSlant`.
Setting a slanted small caps font also set the upright small caps font.

    \documentclass{article}
    \usepackage{fontspec}
    \setmainfont{Latin Modern Roman}[
      UprightFeatures = {
        SmallCapsFont = lmromancaps10-regular,
      },
      ItalicFeatures = {
        SmallCapsFont = lmmonocaps10-oblique,
      },
      SlantedFont = * Slanted,
      SlantedFeatures = {
        SmallCapsFont = lmromancaps10-oblique,
      },
    ]
    \begin{document}
    \textsc{foo} \textit{\textsc{foo}} \textsl{\textsc{foo}}
    \end{document}

The first and third words are in the same font.

**Comment by wspr** [comment]
I believe these problems are now fixed as of `v2.5` (to be released shortly).
